storage/innobase/lock/lock0lock.cc

```diff
@@ -5885,9 +5885,9 @@ namespace Deadlock
 
     if (current_trx && !lock_sys.wr_lock_try())
     {
-      mysql_mutex_unlock(&lock_sys.wait_mutex);
+      lock_sys.wait_mutex_unlock();
       lock_sys.wr_lock(SRW_LOCK_CALL);
-      mysql_mutex_lock(&lock_sys.wait_mutex);
+      lock_sys.wait_mutex_lock();
```

Comment:
We are holding exclusive `lock_sys.latch` here. I do not think that spinning here will help.